### PR TITLE
CLOUD-1310 adds rds cluster-related resources

### PR DESCRIPTION
**Pull request by octopusx**
This commit enables aws-nuke to recognise DBClusterParameterGroup and
DBCluster type resources and remove them.
The changes are necessary to nuke Aurora clusters.

@rebuy-de/prp-aws-nuke please review

### CODE REVIEW

File: resources/rds-dbclusterparametergroups.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	return nil
+}
+
+func (i *RDSDBClusterParameterGroup) Remove() error {
+	params := &rds.DeleteDBClusterParameterGroupInput{
+		DBClusterParameterGroupName: i.name,
+	}
+
+	_, err := i.svc.DeleteDBClusterParameterGroup(params)
+	if err != nil {
```
Comment (bethge) [inline resources/rds-dbclusterparametergroups.go]: Couldn't you just skip this `if` and directly: `return err`

Comment (stephanlindauer) [reply resources/rds-dbclusterparametergroups.go]: IMHO the logic is more understandable and separated like this...
first: error handling
then: making the function return what is should

Comment (svenwltr) [reply resources/rds-dbclusterparametergroups.go]: IMO it looks like unnecessary code :>

Comment (octopusx) [reply resources/rds-dbclusterparametergroups.go]: You could, but then you are implying that whoever looks at the code afterwards understands that `err` defaults to `nil`. The if statement makes it more explicit and easier to read what's going on. Also, this is how that error is handled across all the classes in aws-nuke where it's applicable, so it is following a common pattern in the program.

Comment (octopusx) [reply resources/rds-dbclusterparametergroups.go]: Decision?

Comment (svenwltr) [reply resources/rds-dbclusterparametergroups.go]: Your call

Comment (octopusx) [reply resources/rds-dbclusterparametergroups.go]: I will leave it in for consistency's sake. If this is a big enough problem then it deserves its own pull request and that if statement should be removed everywhere across the whole application.

Comment (bethge) [reply resources/rds-dbclusterparametergroups.go]: Very much agree. Greatly in favor of consistency.

**Pull request merged**